views/templates/pages/index.tpl

```diff
@@ -0,0 +1,14 @@
+<div class="container <?= get_data('cls'); ?>"<?php
+    foreach(get_data('data') as $name => $value) {
+        echo ' data-' . $name . '="' .(is_string($value) ? $value : _dh(json_encode($value))) . '"';
```

Comment:
I disagree on this point as the controller must only provide pure data to the view. The controller must not pre-render HTML code. Eventually we can convert the data to JSON directly from the controller, but we loose the data structure then, this may be a concern for potential view change.